app/screens/channel_bookmark/edit.tsx

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Alert, View, type AlertButton} from 'react-native';
+import {SafeAreaView, type Edge} from 'react-native-safe-area-context';
+
+import {addRecentReaction} from '@actions/local/reactions';
+import {deleteChannelBookmark, editChannelBookmark} from '@actions/remote/channel_bookmark';
+import Button from '@components/button';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import useAndroidHardwareBackHandler from '@hooks/android_back_handler';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import {buildNavigationButton, dismissModal, setButtons} from '@screens/navigation';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import BookmarkDetail from './components/bookmark_detail';
+import AddBookmarkFile from './components/bookmark_file';
+import BookmarkLink from './components/bookmark_link';
+
+import type {AvailableScreens} from '@typings/screens/navigation';
+
+type Props = {
+    bookmark: ChannelBookmark;
+    file?: FileInfo;
+    canDeleteBookmarks: boolean;
+    closeButtonId: string;
+    componentId: AvailableScreens;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    content: {
+        flex: 1,
+        paddingHorizontal: 20,
+        paddingBottom: 16,
+    },
+    progress: {
+        alignItems: 'center',
+        backgroundColor: 'rgba(0, 0, 0, 0.1)',
+        borderRadius: 4,
+        paddingLeft: 3,
+        marginTop: 12,
+    },
+    deleteBg: {backgroundColor: changeOpacity(theme.errorTextColor, 0.16)},
+    deleteContainer: {paddingTop: 32},
+    deleteText: {color: theme.errorTextColor},
+}));
+
+const RIGHT_BUTTON = buildNavigationButton('edit-bookmark', 'channel_bookmark.edit.save_button');
+const edges: Edge[] = ['bottom', 'left', 'right'];
+
+const ChannelBookmarkEdit = ({bookmark: original, canDeleteBookmarks, componentId, closeButtonId, file: originalFile}: Props) => {
+    const {formatMessage} = useIntl();
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+    const styles = getStyleSheet(theme);
+    const [bookmark, setBookmark] = useState<ChannelBookmark>(original);
+    const [file, setFile] = useState<ExtractedFileInfo|undefined>(originalFile);
+    const [isSaving, setIsSaving] = useState(false);
+
+    const enableSaveButton = (enabled: boolean) => {
+        setButtons(componentId, {
+            rightButtons: [{
+                ...RIGHT_BUTTON,
+                color: theme.sidebarHeaderTextColor,
+                text: formatMessage({id: 'channel_bookmark.edit.save_button', defaultMessage: 'Save'}),
+                enabled,
+            }],
+        });
+    };
+
+    const handleError = (error: string, buttons?: AlertButton[]) => {
```

Comment:
This is used in some callbacks, so changes in `formatMessage` or `enableSaveButton` (which in turn depends on `theme`) may not be properly propagated. 0/5 since theme or intl changing is not common in this context.